src/v/storage/disk_log_impl.cc

```diff
@@ -1750,28 +1750,50 @@ disk_log_impl::estimate_reclaim_size(compaction_config cfg) {
         } else if (seg->offsets().dirty_offset <= max_collectible) {
             available_segments.push_back(seg);
         } else {
-            break;
+            remaining_segments.push_back(seg);
         }
     }
 
-    auto [retention, available] = co_await ss::when_all_succeed(
+    auto [retention, available, remaining] = co_await ss::when_all_succeed(
       // reduce segment subject to retention policy
       ss::map_reduce(
         retention_segments,
         [](const segment_set::type& seg) { return seg->persistent_size(); },
-        size_t(0),
-        [](size_t acc, usage u) { return acc + u.total(); }),
+        usage{},
+        [](usage acc, usage u) { return acc + u; }),
 
       // reduce segments available for reclaim
       ss::map_reduce(
         available_segments,
         [](const segment_set::type& seg) { return seg->persistent_size(); },
-        size_t(0),
-        [](size_t acc, usage u) { return acc + u.total(); }));
+        usage{},
+        [](usage acc, usage u) { return acc + u; }),
+
+      // reduce segments available for reclaim
```

Comment:
This comment should say something like: `reduce segments not available for reclaim`.